react/features/recent-list/functions.js

```diff
@@ -71,6 +71,18 @@ export function getRecentRooms(list: Array<Object>): Array<Object> {
     return recentRoomDS.reverse();
 }
 
+/**
+ * Converts the redux subtree into a ready-to-use recent-list Array,
+ * considering the unflattened implementation as well.
+ * FIXME: This may be removed later.
+ *
+ * @param {Array<Object> | Object} state - The redux subtree of the feature.
+ * @returns {Array<Object>}
+ */
+export function toRecentList(state: Array<Object> | Object) {
+    return Array.isArray(state) ? state : state.list || [];
+}
```

Comment:
The same concern about legacy support for the purposes of an internal release goes here, right? We do have a function dedicated to legacy support for a public release so it could've been simpler to not support an internal release.

Reply:
That's not even in an internal release, right? So it's legacy support for something a developer built on their local machine?